uk.co.compucorp.civicrm.hrcore/hrcore.php

```diff
@@ -26,18 +27,9 @@ function hrcore_civicrm_config(&$config) {
  * @param array $tasks
  */
 function hrcore_civicrm_searchTasks($objectName, &$tasks) {
-  if ($objectName !== 'contact' || !CRM_Core_Permission::check('create users')) {
-    return;
+  if (ContactFormSearchTaskAdder::shouldAdd($objectName)) {
+    ContactFormSearchTaskAdder::add($tasks);
   }
```

Comment:
I'm not crazy about instantiating those classes on every request either, I don't usually like static calls but I thought they were more suitable here since they shouldn't have any dependencies or internal state.